machine.js

```diff
@@ -129,15 +138,33 @@ chokidar.watch("projects.json").on("change", () => {
 
 /* Global max capacity */
 var maxCapacity = 1;
+var GPUsCapacity = [];
+var GPUCapacity = 1;
+
+// Checks if GPU resources are required
+var isGPURequired = function (projId) {
+  return "gpu_capacity" in projects[projId];
+};
 
 var getCapacity = function(projId) {
   var capacity = 0;
   if (projects[projId]) {
     capacity = Math.floor(maxCapacity / projects[projId].capacity);
+    if (isGPURequired(projId)) {
+      // if we have any GPUs 
+      if (('gpus' in specs) && (specs.gpus.length > 0)) {
+        var gpu_available_capacity = GPUsCapacity.reduce((a, b) => a + b, 0);
+        var gpu_capacity = Math.floor (gpu_available_capacity / projects[projId].gpu_capacity);
```

Comment:
I think afterwards you need to set `capacity` to the minimum of `capacity` and `gpu_capacity`, so that the true minimum is returned.